core/trino-main/src/main/java/io/trino/sql/SessionSpecificationInterpreter.java

```diff
@@ -0,0 +1,113 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.sql;
+
+import com.google.common.collect.HashBasedTable;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.Table;
+import io.trino.sql.tree.BooleanLiteral;
+import io.trino.sql.tree.DecimalLiteral;
+import io.trino.sql.tree.DoubleLiteral;
+import io.trino.sql.tree.Expression;
+import io.trino.sql.tree.Literal;
+import io.trino.sql.tree.LongLiteral;
+import io.trino.sql.tree.NullLiteral;
+import io.trino.sql.tree.Query;
+import io.trino.sql.tree.SessionSpecification;
+import io.trino.sql.tree.Statement;
+import io.trino.sql.tree.StringLiteral;
+
+import java.util.List;
+import java.util.Map;
+
+import static io.trino.spi.StandardErrorCode.INVALID_SESSION_PROPERTY;
+import static io.trino.sql.analyzer.SemanticExceptions.semanticException;
+import static java.util.Objects.requireNonNull;
+
+public class SessionSpecificationInterpreter
+{
+    private SessionSpecificationInterpreter() {}
+
+    public static ResolvedSessionSpecifications evaluate(Statement statement)
+    {
+        if (statement instanceof Query queryStatement) {
+            return evaluate(queryStatement.getSessionProperties());
+        }
+
+        return new ResolvedSessionSpecifications(ImmutableMap.of(), ImmutableMap.of());
+    }
+
+    private static ResolvedSessionSpecifications evaluate(List<SessionSpecification> specifications)
```

Comment:
Use `SessionPropertyManager.evaluatePropertyValue()` for this. Session properties support full expressions, not just literals. See `SetSessionTask` for how this should work.